vue-app/src/plugins/Web3/index.ts

```diff
@@ -111,6 +120,15 @@ export default {
       }
 
       plugin.provider = null
+      plugin.user = null
+    }
+
+    // If previous provider was found, initiate connection.
+    if (alreadyConnectedProvider) {
+      const response = plugin.connectWallet(alreadyConnectedProvider)
+      if (response.state !== 'fulfilled') {
```

Comment:
Wondering what `state` does in this case since we are not `await`ing the promise. I mean, in which other cases are we going to face a different value than `'fulfilled'`?

Reply:
@pettinarip So this only returns `'fulfilled'` is the user actually signs... Adding this basically is checking if the user cancels the prompt. If they do, it will then clear the localStorage.

ie: If the user clicks "cancel" on MetaMask -> returns `response.state = 'pending'`... 

When I tried to wrap this in `await` syntax, the app does not respond and update with the logged in user. Clicking "Connect" and signing just leads to the same screen with the "Connect" button still showing. Keeping it synchronous as above seems to work properly.

Reply:
I just tested this and understood what you want to do.

Actually, checking the promise `state` prop is not going to tell you if the user has accepted or not the prompt. That state is part of the Promise lifecycle (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). When the Promise is `resolve`d or `reject`ed, the state variable is going to change.

So, in this case, `response.state` is always going to be `pending` because we are not waiting for it to end. Basically, the condition is always going to be true.

Regarding what you want to achieve here I think we could do the following:
```ts
if (alreadyConnectedProvider) {
  // What we really want to do is to clean the `alreadyConnectedProvider` provider, not do a complete disconnection.
  // This is mainly to avoid future problems if we add more logic in the `disconnectWallet` method
  lsRemove(CONNECTED_PROVIDER) 
  plugin.connectWallet(alreadyConnectedProvider)
}
```

Reply:
Agreed - thanks for the explanation @pettinarip!